### ISSUE TITLE
HTML entity handling

### ISSUE THREAD

**Issue by notenoughneon** [glennjones/microformat-node#27]
HTML entities are indistinguishable from actual tags in the "html" part of e- properties. Additionally, value and p- properties may be corrupted when using textFormat: whitespacetrimmed.

Test case:
```html
<div class="h-entry">
    <div class="p-name e-content">x&lt;y AT&amp;T &lt;b&gt;NotBold&lt;/b&gt; <b>Bold</b></div>
</div>
```
Output (textformat: 'whitespacetrimmed'):
```json
{
    "items": [{
        "type": ["h-entry"],
        "properties": {
            "name": ["xNotBold Bold"],
            "content": [{
                "value": "xNotBold Bold",
                "html": "x<y AT&T <b>NotBold</b> <b>Bold</b>"
            }]
        }
    }],
    "rels": {},
    "rel-urls": {}
}
```
Output (textFormat: 'normalised'):
```json
{
    "items": [{
        "type": ["h-entry"],
        "properties": {
            "name": ["x<y AT&T <b>NotBold</b> Bold"],
            "content": [{
                "value": "x<y AT&T <b>NotBold</b> Bold",
                "html": "x<y AT&T <b>NotBold</b> <b>Bold</b>"
            }]
        }
    }],
    "rels": {},
    "rel-urls": {}
}
```
Expected for both cases:
```json
{
    "items": [{
        "type": ["h-entry"],
        "properties": {
            "name": ["x<y AT&T <b>NotBold</b> Bold"],
            "content": [{
                "value": "x<y AT&T <b>NotBold</b> Bold",
                "html": "x&lt;y AT&amp;T &lt;b&gt;NotBold&lt;/b&gt; <b>Bold</b>"
            }]
        }
    }],
    "rels": {},
    "rel-urls": {}
}
```

**Comment by glennjones** [comment]
Hi 
Sorry to hear your having issues with the parsing. I am working on a few other issue on other open source projects. I will try and look at this within the next couple of weeks.

**Comment by notenoughneon** [comment]
Thanks for taking a look at this. I looked at https://github.com/glennjones/microformat-shiv/blob/master/lib/html.js to see if it was anything I could fix, but I don't have a good understanding of the data-include use case.

**Comment by glennjones** [comment]
I am very sorry its taken me so long to fix this, but its now sorted in v2.0.0 I have added a test to the microformats test suite https://github.com/microformats/tests/blob/master/tests/microformats-v2/h-entry/encoding.html based on your example. 

This project uses the same codebase as https://github.com/glennjones/microformat-shiv which is used as an internal component in the Firefox browser. The process of making changes is a little more time consuming because of this and I had to clear a couple of days to get this done.

Thanks for pointing out the problem, any others issue please get in touch
